Modules/ECG-Baseline/ECG_Baseline.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+#include <random>
+#include <algorithm>
+#include <iterator>
+#include <vector>
+#include <cmath>
+#include <fstream>
+#include <utility>
+
+# define M_PIl          3.141592653589793238462643383279502884L
+
+enum ConvolutionType
+{
+	Same = 0,
+	Full = 1,
+}; 
+
+
+struct MovingAverageFilterParameters
+{
+	unsigned int windowLenght = 10;
+	std::vector<float> convolutionWindow = {};
+	std::vector<float> filteredSignal;
+};
+
+struct LMSFilterParameters
+{
+	float convergenceRate = 0.2f;
+	std::vector<float> weights = { 0.0, 0.0, 0.0, 0.0, 0.0 };
+	std::vector<float> errors = {};
+	std::vector<float> desireSignalCoefficients = { 1, 0.5, 0.25,0.125, 0.0625 };
+	std::vector<float> desireSignal = {};
+	std::vector<float> currentlyCalculatedSubvector = {};
+	std::vector<float> filteredSignal;
+};
+
+struct ButterworthFilterParameters
+{
+	unsigned int fLowPass = 15;
+	unsigned int fHighPass = 5;
+	unsigned int fSampling = 360;
```

Comment:
Czy wszystkie sygnały wfdb mają taką częstotliwość? To informacja z pliku `*.hea` i może się różnić.

Reply:
done